packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -44,6 +44,14 @@ export interface IProject extends IResource, iam.IGrantable, ec2.IConnectable {
   /** The IAM service Role of this Project. Undefined for imported Projects. */
   readonly role?: iam.IRole;
 
+  /** The batch IAM service Role of this Project. Undefined for imported Projects or Projects that have not configured batch builds. */
+  readonly batchRole?: iam.IRole;
```

Comment:
> Also I made the batchRole a public readonly property because this is then inline with the existing role property, and because of that thought enableBatchBuilds() didn't really need to return anything.

I would get rid of this, because the value of this property changes with the lifetime of the Project instance. For this reason, I'd prefer for `enableBatchBuilds()` to return a value, like I originally suggested.

Reply:
Cool will update 👍  It felt weird because if you wanted to do something with the batch role, if batch builds were enabled, you wouldn't be able to do that if `enableBatchBuilds()` returned the thing you needed, because the act of calling it enables it.